extensions/cloud/portability-cloud-microsoft/src/main/java/org/dataportabilityproject/cloud/microsoft/cosmos/AzureKeyVaultStore.java

```diff
@@ -0,0 +1,57 @@
+package org.dataportabilityproject.cloud.microsoft.cosmos;
+
+import com.microsoft.azure.AzureEnvironment;
+import com.microsoft.azure.credentials.ApplicationTokenCredentials;
+import com.microsoft.azure.keyvault.KeyVaultClient;
+import com.microsoft.azure.keyvault.models.SecretBundle;
+import org.dataportabilityproject.spi.cloud.storage.AppCredentialStore;
+import org.dataportabilityproject.types.transfer.auth.AppCredentials;
+
+import java.io.IOException;
+import java.util.Objects;
+
+import static java.lang.System.getenv;
+
+/**
+ * A secrets store backed by Azure Key Vault.
+ *
+ * <p>
+ */
+public class AzureKeyVaultStore implements AppCredentialStore {
+  private static final String VAULT_ADDRESS = "https://%s.vault.azure.net/";
+
+  private final String vaultUrl;
+  private final KeyVaultClient vaultClient;
+
+  public AzureKeyVaultStore(String keyVaultName) {
+    vaultUrl = String.format(VAULT_ADDRESS, keyVaultName);
+
+    String azureClientId = getenv("AZURE_CLIENT_ID");
```

Comment:
They don't but my thinking was they would be supplied by whatever controls the process since they are sensitive. For example, they could be a Kubernetes secret. We could have it sourced from props and override to env but that seems potentially dangerous. What do you think?

Reply:
OK cool, that sounds good. Maybe just add a comment reflecting that this should come from somewhere secure like a Kubernetes secret, so someone writing a distribution knows to do that and not set a raw cred in as an env variable in the container, for example.